[PATCH v3] linux-generic: pool: validate params before creating pool

Bill-Fischofer-Linaro opened the pull request:
Enhance the check_params() routine to verify that params != NULL
prior to dereferencing it.

This fixes Bug https://bugs.linaro.org/show_bug.cgi?id=3465

Signed-off-by: Bill Fischofer <bill.fischofer@linaro.org>

muvarov commented:
@Bill-Fischofer-Linaro since you touched  check_params() can you also change all printf to ODP_PRINT()  there or even ODP_ERR might be better choose for it.  Also it looks like check_params() is very verbose so it's better separate to have separate check with print for params and  capabilities.

Bill-Fischofer-Linaro commented:
Changes for v2:
* Rebased to latest master
* Incorporated @muvarov review comment to avoid direct use of `printf()`. Use `ODP_DBG()` in `check_params()` and `ODP__PRINT()` in `odp_pool_print()`.

Bill-Fischofer-Linaro commented:
Changes for v3:
* Fixed checkpatch issue in Part 2.

muvarov commented:
Merged.
